rs/ic_os/hostos_tool/src/main.rs

```diff
@@ -62,27 +62,20 @@ pub fn main() -> Result<()> {
             eprintln!("Using config: {:?}", config_map);
 
             let network_info = NetworkInfo::from_config_map(&config_map)?;
-            eprintln!("Network info config: {:?}", &network_info);
 
-            let deployment = read_deployment_file(Path::new(&opts.deployment_file));
+            let mut network_settings = network_info.to_network_settings();
+            eprintln!("Network settings config: {:?}", &network_settings);
 
-            let deployment_name: Option<&str> = match &deployment {
-                Ok(deployment) => Some(deployment.deployment.name.as_str()),
-                Err(e) => {
-                    eprintln!("Error retrieving deployment file: {e}. Continuing without it");
-                    None
-                }
-            };
+            let deployment = read_deployment_file(Path::new(&opts.deployment_file))?;
 
-            let mgmt_mac: Option<&str> = match &deployment {
-                Ok(deployment) => deployment.deployment.mgmt_mac.as_deref(),
-                Err(_) => None,
-            };
+            // TODO: NODE-1466: Remove in configuration revamp (HostOS and GuestOS integration).
+            // Once HostOS is using the config struct, all config will be contained there
+            // and we won't need to read mgmt_mac from depolyment directly.
+            network_settings.mgmt_mac = deployment.deployment.mgmt_mac;
```

Comment:
@frankdavid In response to your comment [here](https://github.com/dfinity/ic/pull/1365#discussion_r1752531928)
> When can this be an error? Could we fail or at least log the error?

I've removed the match statement in this PR now that I've added the mgmt_mac to the new config structure. Unfortunately, because I moved mgmt_mac from config.ini to deployment.json, we're temporarily in an awkward state where we need to update network_settings.mgmt_mac. It's a bit complex, but the good news is that eventually, mgmt_mac will always be contained in the config struct.

To your question, mgmt_mac is strictly a testing setting. So we log the whole config on HostOS/GuestOS boot, but we won't have to log it here.

Reply:
Nevermind! I've refactored things and it's cleaner now.